backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/ddl/public/functions/sync_translation_of_tables_without_ad_element_id.sql

```diff
@@ -0,0 +1,176 @@
+                                            || 'SELECT base_table_id, bt.ad_client_ID, bt.ad_org_id, now(), 0, now(), 0, ''N'', bt.IsActive, %s '
+                                            || 'FROM base_table bt '
+                                            || 'WHERE bt.base_table_id= %s '
+                                            || 'AND NOT EXISTS (SELECT 1 FROM base_table_trl btt WHERE btt.base_table_id = %s AND btt.ad_language = ''%s'')';
+    update_query_template       varchar := 'WITH X AS (SELECT * FROM base_table_trl WHERE base_table_id = %s AND ad_language =  ''%s'') '
+                                            || 'UPDATE base_table as bt '
+                                            || 'SET %s '
+                                            || 'FROM X '
+                                            || 'WHERE bt.base_table_id = x.base_table_id ';
+BEGIN
+
+    SELECT ad_language INTO base_language FROM ad_language WHERE isbaselanguage = 'Y';
+
+    IF (base_language IS NULL OR TRIM(base_language) = '') THEN
+        RAISE EXCEPTION 'Base language is not set';
+    END IF;
+
+    FOR table_name IN SELECT UPPER(tablename) FROM ad_table WHERE UPPER(tablename) LIKE UPPER('%_Trl') AND UPPER(tablename) NOT LIKE ALL( IGNORED_TABLES ) ORDER BY 1
+        LOOP
+            base_table := SUBSTRING(table_name, 1, LENGTH(table_name) - 4); -- that means without _TRL. e.g. AD_Ref_List
+            insert_query_set_template := '';
+            update_query_set_template := '';
+            update_query_and_template := '';
+            FOR v_columnname IN SELECT columnname FROM ad_column WHERE istranslated = 'Y' AND ad_table_id = get_table_id(base_table) -- e.g. AD_Ref_List.Name
+                LOOP
+                    insert_query_set_template := insert_query_set_template || FORMAT('%s ,', v_columnname); -- Name ,Description ,
+                    update_query_set_template := update_query_set_template || FORMAT('%s = x.%s, ', v_columnname, v_columnname); -- Name = x.Name ,Description = x.Description ,
+                    update_query_and_template := update_query_and_template || FORMAT('bt.%s <> x.%s OR ', v_columnname, v_columnname); -- bt.Name <> x.Name OR bt.Description <> x.Description OR
+                END LOOP;
+
+            insert_query_set_template := insert_query_set_template ; --=> Name ,Description,
+            update_query_set_template := SUBSTRING(update_query_set_template, 1, LENGTH(update_query_set_template) - 2); -- delete the last ", " => Name = x.Name ,Description = x.Description
+            update_query_and_template := 'AND ( ' || SUBSTRING( update_query_and_template, 1, LENGTH(update_query_and_template) - 3) || ' );'; -- delete the last "OR " => AND ( Name <> x.Name OR Description <> x.Description )
+
+            FOR base_table_id IN EXECUTE 'SELECT ' || base_table || '_ID FROM ' || base_table -- e.g. AD_Ref_List_ID = 123456
```

Comment:
why do we have to do it for each record? why not mass update all records?

imagine we have M_Product with 10 mio records. How much would take to insert/update into M_Product_Trl?

Reply:
i couldn't find a solution to do it as mass update, because of the subqueries:
```sql
AND NOT EXISTS(SELECT 1
                                             FROM AD_Ref_List_trl btt
                                             WHERE btt.AD_Ref_List_id = trl.AD_Ref_List_id
                                               AND btt.ad_language = 'en_US');

WITH X AS (SELECT * FROM AD_Ref_List_trl WHERE AD_Ref_List_id = 123456 AND ad_language =  'en_US')
```

Reply:
@mrie04  can u pls paste me how an SQL looks like (an example) 
and i will check if I could rewrite it as a mass update SQL
?

Reply:
@teo it's already included as comment inside the function